### PR TITLE
Merge changes from rss dart fork.

### PR DESCRIPTION
**Pull request by amugofjava**
Hi @matthew-carroll,

This PR contains two commits which merges in the changes from the fork myself and @Feichtmeier have been working on.

The first commit brings the `intl` and `xml` dependencies up to date and migrates the XML code. The second adds support for the Podcasting 2.0 persons tag with relevant tests and improves the Dart docs for the Podcasting 2.0 classes.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@amugofjava is there a reason that all 3 of those things need to be done in a single PR? Could we instead handle those as 3 PRs, referencing 3 issues, and keep the clarity in the review and commit history?

**Comment by amugofjava** [comment]
@matthew-carroll - this PR relates to the comments here [#20](https://github.com/Flutter-Bounty-Hunters/dart-rss/pull/20) which I just saw as a single PR split into two commits.

I am happy to split the PR and link to issues, though if by three that's separate PRs for `intl` and `xml` that will be a little fiddler.

**Comment by matthew-carroll** [comment]
I'll review this all as one PR so that you don't have to rework it. In the future, I would prefer issue tickets filed for "upgrading xml", "upgrading intl", and "add Podcasting 2.0 person", and then separate PRs for each.

File: lib/domain/podcast_index/rss_podcast_index_chapters.dart [hunk 1]
```diff
@@ -1,7 +1,14 @@
 import 'package:xml/xml.dart';
 
+/// Episodes that support the Podcasting 2.0 namespace can have a list of
```
Comment (matthew-carroll) [inline lib/domain/podcast_index/rss_podcast_index_chapters.dart:3]: In keeping with Dart Doc practices, please add a blank line between the first sentence and whatever follows. And please adjust any other Dart Docs in this PR, accordingly.

File: lib/domain/rss_item_itunes.dart [hunk 2]
```diff
@@ -38,31 +38,31 @@ class RssItemItunes {
...
-              ?.text
+              ?.innerText
               .split(',')
               .map((keyword) => keyword.trim())
               .toList() ??
           const <String>[],
       image: RssItunesImage.parse(findElementOrNull(element, 'itunes:image')),
       category: RssItunesCategory.parse(
-          findElementOrNull(element, 'itunes:category')),
+        findElementOrNull(element, 'itunes:category'),),
```
Comment (matthew-carroll) [inline lib/domain/rss_item_itunes.dart]: Looks like there's a formatting issue here - looks like this comma should be followed by a line break, or if you don't want a line break, then there shouldn't be a comma.

**Review by matthew-carroll — APPROVED** [review]
LGTM

**Comment by NomadicDeveloper22** [comment]
what's the ETA on pub.dev getting this update?

**Comment by matthew-carroll** [comment]
I just released v3.0.1

**Pull request merged**